public/components/trace_analytics/components/traces/span_detail_flyout.tsx

```diff
@@ -74,6 +74,31 @@ const getSpanValue = (span: object, mode: TraceAnalyticsMode, field: SpanField)
   return get(span, fieldKey);
 };
 
+// Flatten nested objects with dot notation
+const flattenObject = (
+  obj: any,
+  prefix = '',
+  result: Record<string, any> = {}
+): Record<string, any> => {
+  for (const key in obj) {
+    if (obj.hasOwnProperty(key)) {
+      const newKey = prefix ? `${prefix}.${key}` : key;
+
+      if (obj[key] !== null && typeof obj[key] === 'object' && !Array.isArray(obj[key])) {
+        flattenObject(obj[key], newKey, result);
+      } else {
+        result[newKey] = obj[key];
+      }
+    }
+  }
+  return result;
+};
+
+// Get all span attributes including nested ones
+const getAllSpanAttributes = (span: any): Record<string, any> => {
```

Comment:
do we need this wrapper function is all it is doing is flatten?

Reply:
Removed, should be clearer by calling it directly:
```
const allAttributes = flattenObject(span);
```